internal/dag/gatewayapi_processor.go

```diff
@@ -896,30 +895,48 @@ func (p *GatewayAPIProcessor) validateBackendRef(backendRef gatewayapi_v1alpha2.
 	return service, nil
 }
 
-func gatewayPathMatchCondition(match *gatewayapi_v1alpha2.HTTPPathMatch) (MatchCondition, error) {
-
+func gatewayPathMatchCondition(match *gatewayapi_v1alpha2.HTTPPathMatch, routeAccessor *status.RouteConditionsUpdate) (MatchCondition, bool) {
 	if match == nil {
-		return &PrefixMatchCondition{Prefix: "/"}, nil
+		return &PrefixMatchCondition{Prefix: "/"}, true
 	}
 
 	path := pointer.StringDeref(match.Value, "/")
 
 	// If path match type is not defined, default to 'PathPrefix'.
 	if match.Type == nil || *match.Type == gatewayapi_v1alpha2.PathMatchPathPrefix {
+		if !strings.HasPrefix(path, "/") {
+			routeAccessor.AddCondition(status.ConditionValidMatches, metav1.ConditionFalse, status.ReasonInvalidPathMatch, "Match.Path.Value must start with '/'.")
```

Comment:
note, this is a Contour-specific condition type for now as there is not one defined upstream.